src/VirtualClient/VirtualClient.Actions/SysbenchOLTP/SysbenchOLTPServerExecutor.cs

```diff
@@ -46,7 +46,17 @@ protected override async Task InitializeAsync(EventContext telemetryContext, Can
         {
             await base.InitializeAsync(telemetryContext, cancellationToken).ConfigureAwait(false);
             this.InitializeApiClients();
-            this.ConfigureMySQLPrivileges(cancellationToken);
+
+            SysbenchOLTPState state = await this.StateManager.GetStateAsync<SysbenchOLTPState>($"{nameof(SysbenchOLTPState)}", cancellationToken)
+                ?? new SysbenchOLTPState();
+
+            if (!state.MySQLDBInitialized)
```

Comment:
Could you add unit test to verify the executor would do the right thing for different state? You could mock the state object and verify the SQL will be configured if not initialized, and will be skipped if initialized.